### PR TITLE
build: update to go 1.15

### PR DESCRIPTION
**Pull request by jpwhitemn**
Updated Dockerfile, go.mod for Go 1.15

Removed go reference in Jenkinsfile so it defaults to CI/CD pipeline default

Updated README Go version references

Fixes: #284
Signed-off-by: Jim White <jpwhite_mn@yahoo.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x ] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x ] Other... Please describe:  upgrade to Go 1.15

## What is the current behavior?

n/a

Issue Number: #284 

## What is the new behavior?
none

## Does this PR introduce a breaking change?

- [ ] Yes
- [ x] No

## Other information

### CODE REVIEW

**Review by lenny-intel — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ The default encoding for the models is JSON, although in at least one case --
 YAML encoding is also supported since a device profile is defined as a YAML document.
 
 ### Installation ###
-* Make sure you're using at least Go 1.11.1 and have modules enabled, i.e. have an initialized  go.mod file 
+* Make sure you're using at least Go 1.11.1 (EdgeX currently uses Go 1.15.2) and have modules enabled, i.e. have an initialized  go.mod file
```
Comment (cloudxxx8) [inline README.md:12]: we might just state 1.15, because we don't define the patch number in the go.mod and Dockerfile

Comment (cloudxxx8) [reply README.md:12]: in edgex-go, we are using 1.15.x

**Review by lenny-intel — APPROVED** [review]
LGTM

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**